Mathlib/Analysis/Polynomial/CauchyBound.lean

```diff
@@ -0,0 +1,88 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.Algebra.GeomSum
+import Mathlib.Algebra.Polynomial.Eval
+import Mathlib.Analysis.Normed.Field.Basic
+
+/-!
+# Cauchy's bound on polynomial roots.
+-/
+
+variable {K : Type*} [NormedDivisionRing K]
+
+open Polynomial Finset NNReal
+
+/--
+Cauchy's bound on the roots of a given polynomial.
+-/
+noncomputable def cauchyBound (p : K[X]) : ℝ≥0 :=
```

Comment:
Perhaps worth having trivial simp lemmas for `0`, `1`, `C`, and `X`? Having one for `smul` might be more interesting too.

Reply:
Added